spec.emu

```diff
@@ -47,24 +27,54 @@ contributors: Jordan Harband
         <emu-alg>
           1. Let _str_ be ? ToString(_S_).
           1. Let _cpList_ be StringToCodePoints(_str_).
-          1. Let _punctuators_ be the following String, which consists of every ASCII punctuator except U+005F (LOW LINE): *"(){}[]|,.?\*+-^$=<>\/#&!%:;@~'"`"*.
-          1. Let _toEscape_ be StringToCodePoints(_punctuators_).
           1. Let _escapedList_ be a new empty List.
           1. For each code point _c_ in _cpList_, do
             1. If _escapedList_ is empty and _c_ is matched by |DecimalDigit|, then
               1. Append code unit U+005C (REVERSE SOLIDUS) to _escapedList_.
               1. Append code unit U+0078 (LATIN SMALL LETTER X) to _escapedList_.
               1. Append code unit U+0033 (DIGIT THREE) to _escapedList_.
-            1. Else if _toEscape_ contains _c_ or _c_ is matched by |WhiteSpace|, then
-              1. Append code unit U+005C (REVERSE SOLIDUS) to _escapedList_.
-            1. Append _c_ to _escapedList_.
+              1. Append _c_ to _escapedList_.
+            1. Else,
+              1. Append the code units in EncodeForRegExpEscape(_c_) to _escapedList_.
           1. Return CodePointsToString(_escapedList_).
         </emu-alg>
 
         <emu-note>
           <p>`escape` takes a string and escapes it so it can be literally represented as a pattern. In contrast EscapeRegExpPattern (as the name implies) takes a pattern and escapes it so that it can be represented as a string. While the two are related, they do not share the same character escape set or perform similar actions.</p>
         </emu-note>
       </emu-clause>
+
+      <emu-clause id="sec-encode" type="abstract operation">
+        <h1>
+          EncodeForRegExpEscape (
+            _c_: a code unit,
```

Comment:
We're iterating `For each code point _c_ in _cpList_ ` code points (roughly a u32) and this can only handle code units (u16) due to the `The length of _hex_ is at most 4` assertion. We could iterate code units since we only care to escape ASCII chars, or we need to update this to handle `\u{123456}` sequences.

Reply:
The latter. This should take a code point as input.